PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java

```diff
@@ -249,19 +298,8 @@ public void menuAboutToShow(IMenuManager manager) {
                     return;
                 }
                 if (tableViewer.getSelection() instanceof IStructuredSelection) {
-                    Action action = new Action("Open") {
-                        @Override
-                        public void run() {
-                            viewMessageText();
-                        }
-                    };
-                    manager.add(action);
-                    action = new Action("Dequeue") {
-                        @Override
-                        public void run() {
-                            dequeueFirstMessage();
-                        }
-                    };
+                    manager.add(fileEditorVirtualNode.createPopupAction(OPEN));
+                    Action action = fileEditorVirtualNode.createPopupAction(DEQUEUE);
```

Comment:
`fileEditorVirtualNode` should NEVER be null. It's instantiated in the `QueueFileEditor`'s contructor.